### ISSUE TITLE
AD Theme Direct WFS 2.0 Test - No Temporal values found

### ISSUE THREAD

**Issue by robblis** [opengeospatial/ets-wfs20#185]
To the OGC Community,

The following is the error:
 
OGC TEAM Engine reported a failed test: No temporal values found in results: property is "http://inspire.ec.europa.eu/schemas/ad/4.0 [inspire.ec.europa.eu]":beginLifespanVersion. expected [false] but found [true]
 
This is the decoded request that the validator sends
 
http://agsinspire1.esri.com/arcgis/rest/services/ADTest/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns63:ThoroughfareName&namespaces=xmlns(xml,http://www.w3.org/XML/1998/namespace),xmlns(ns63,http://inspire.ec.europa.eu/schemas/ad/4.0),xmlns(wfs,http://www.opengis.net/wfs/2.0)&filter=<fes:Filter xmlns:ns63="http://inspire.ec.europa.eu/schemas/ad/4.0 [inspire.ec.europa.eu]" xmlns:wfs="http://www.opengis.net/wfs/2.0 [opengis.net]" xmlns:fes="http://www.opengis.net/fes/2.0 [opengis.net]"><fes:During><fes:ValueReference xmlns:tns="http://inspire.ec.europa.eu/schemas/ad/4.0 [inspire.ec.europa.eu]">tns:beginLifespanVersion</fes:ValueReference><gml:TimePeriod xmlns:gml="http://www.opengis.net/gml/3.2 [opengis.net]" frame="http://www.iso.org/iso/iso8601 [iso.org]" gml:id="TP01"><gml:beginPosition>1988-09-05T10:00:00Z</gml:beginPosition><gml:endPosition>1988-09-05T12:00:00Z</gml:endPosition></gml:TimePeriod></fes:During></fes:Filter>
 
 
The existing beginLifespanVersion value in the database for the ThoroughfareName entry is: 9/5/1988 6:00:00 AM
If I adjust the request’s gml:beginPosition and gml:endPosition to the following highlighted below, by taking the “Z” off the time and adjusting the beginPosition time an hour before the current value, then I get the returned component.
 
 
http://agsinspire1.esri.com/arcgis/rest/services/ADTest/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns63:ThoroughfareName&namespaces=xmlns(xml,http://www.w3.org/XML/1998/namespace),xmlns(ns63,http://inspire.ec.europa.eu/schemas/ad/4.0),xmlns(wfs,http://www.opengis.net/wfs/2.0)&filter=<fes:Filter xmlns:ns63="http://inspire.ec.europa.eu/schemas/ad/4.0 [inspire.ec.europa.eu]" xmlns:wfs="http://www.opengis.net/wfs/2.0 [opengis.net]" xmlns:fes="http://www.opengis.net/fes/2.0 [opengis.net]"><fes:During><fes:ValueReference xmlns:tns="http://inspire.ec.europa.eu/schemas/ad/4.0 [inspire.ec.europa.eu]">tns:beginLifespanVersion</fes:ValueReference><gml:TimePeriod xmlns:gml="http://www.opengis.net/gml/3.2 [opengis.net]" frame="http://www.iso.org/iso/iso8601 [iso.org]" gml:id="TP01"><gml:beginPosition>1988-09-05T05:00:00</gml:beginPosition><gml:endPosition>1988-09-05T12:00:00</gml:endPosition></gml:TimePeriod></fes:During></fes:Filter>
 

Our service returns datetimes without timezones. This means (as per XML Schema spec) the datetimes are in the servers local timezone. This timezone is unknown for external users, because it is not declared by the service. 

I am not aware that the WFS spec supports declaring the timezone by the service.

Also, it almost seems like the Validator is assuming the datetimes returned by the service are in the local timezone of the validator.

Can someone please assess this issue and give us guidance?

Thank you.

**Comment by dstenger** [comment]
@keshav-nangare Can you please do further investigation?

**Comment by keshav-nangare** [comment]
@robblis 

Can you please provide service to analyze the reported behavior?

**Comment by robblis** [comment]
Hi @keshav-nangare,

The service for this test is running.

http://agsinspire1.esri.com/arcgis/rest/services/ADTest/MapServer/exts/InspireFeatureDownload/service?VERSION=2.0.0&SERVICE=WFS&REQUEST=GetCapabilities&LANGUAGE=&TYPENAMES=&MAXFEATURES=&COUNT=&f=image

Thanks,
Robert

**Comment by dstenger** [comment]
W3C XML Schema Part 2 was checked (also see https://github.com/opengeospatial/ets-wfs20/issues/183#issuecomment-747364942) and states the following:

```
...
3.2.7.3 Timezones
...
The lexical representation of a timezone is a string of the form: (('+' | '-') hh ':' mm) | 'Z', where
...
```

Thus, the specification states that timezoned values are represented by a "Z" and that this is a legal notation.

@robblis So, a service should be capable to handle this notation. Does this answer your question and do you agree with the conclusion?

**Comment by robblis** [comment]
Hi @dstenger,

We finally were able to do some more testing on this.

So, we think the issue is the time conversion to "Z" time is not correct.

So, going back to the test stated above, the time query is: gml:beginPosition1988-09-05T10:00:00Z  and  gml:endPosition1988-09-05T12:00:00Z

The time for the beginLifespanVersion in the ThoroughfareName entry is: 9/5/1988 6:00:00 AM (PDT). This would translate to 13:00:00Z time. So I'm not sure why the generated query is generating the endPosition at 12:00:00Z. How is it determining that particular time?

![image](https://user-images.githubusercontent.com/3453448/113363723-71026000-9306-11eb-8dcb-9a4d5bc63e0c.png)

If I run the same request and change the endPosition to13:01:00Z then a record is returned.

Is this a miscalculation for the time element, or does our time in the database need to be represented as "Z" time?

Thanks,
Robert

**Comment by dstenger** [comment]
Hi @robblis,

Currently, your service is not reachable. Is it possible to enable it again so that we can use it for testing purposes?

If I understand you correctly, your service is supporting time patterns with "Z" but the calculation of the filter values is wrong?
If that is the case, we will check the test suite regarding this behavior (a running test service would help us a lot).

Regards,
Dirk

**Comment by robblis** [comment]
Hi @dstenger,

Which service are you trying to reach?

The service listed above...
http://agsinspire1.esri.com/arcgis/rest/services/ADTest/MapServer/exts/InspireFeatureDownload/service?VERSION=2.0.0&SERVICE=WFS&REQUEST=GetCapabilities&LANGUAGE=&TYPENAMES=&MAXFEATURES=&COUNT=&f=image
...is reachable.

Thanks,
Robert

**Comment by ghobona** [comment]
Hi @robblis ,

The service is not reachable. Could you please check your firewall?

Regards,

Gobe

![Screenshot 2021-12-01 at 10 30 44](https://user-images.githubusercontent.com/18121038/144218512-6ea3c77c-29ff-41af-ae40-2cb5803abefa.png)

**Comment by JohannaOtt** [comment]
@bpross-52n I have a similar issue with [this WFS](https://geodienste.komm.one/ows/services/org.506.78427a5b-2f82-4311-ba7d-74eb7b8c10f4_wfs/org.506.e21f771b-0d06-4dbc-b246-60aa8d12ed52?SERVICE=WFS&REQUEST=GetCapabilities&VERSION=2.0.0).
![grafik](https://user-images.githubusercontent.com/34163838/161543879-05d0ad5a-d7fd-4572-801b-1a05fb7487fe.png)
There are two kinds of error messages:
1. No temporal values found in results: property is "http://inspire.ec.europa.eu/schemas/plu/4.0":validFrom.  --> there is a validFrom value in the dataset in the SpatialPlan object (the OfficialDoument objects do not need one - see [UML](https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/EARoot/EA2/EA3/EA10/EA1/EA4/EA8445.htm)) and it is valid according to the INSPIRE Validator (dataset and report: 
[validFrom.zip](https://github.com/opengeospatial/ets-wfs20/files/8409689/validFrom.zip) )
Could you please check why the error message is occurring?
2. java.lang.AssertionError: t1 (Period: 2012-04-04T00:00:00.000+0000/2012-04-04T23:59:59.999+0000) is not related to t2 (Period: 2012-04-04T16:19:59.999+0000/2012-04-05T00:59:59.998+0000): BEFORE. The actual temporal relation is OVERLAPS -> not sure what t1 and t2 are in this case and what would be invalid about them.

If you want me to open a separate issue on that, please let me know. But as one of the error messages is the same as in this issue, I thought it would make sense to add it here.


Thank you and all the best

Johanna

**Comment by robblis** [comment]
Hi @JohannaOtt,

We've turned on the following service to test against.

https://agsinspire1.esri.com/arcgis/rest/services/ADTest/MapServer/exts/InspireFeatureDownload/service?VERSION=2.0.0&SERVICE=WFS&REQUEST=GetCapabilities&LANGUAGE=&TYPENAMES=&MAXFEATURES=&COUNT=&f=image

Let us know if you need anything else from us.

Thanks,
Robert

**Comment by JohannaOtt** [comment]
@robblis Not sure if that is a misunderstanding - I reported a similar issue as yours as a validator user. But I am sure your example service will be helpful for the validator team.

**Comment by robblis** [comment]
Hi @dstenger 

Sorry, I had posted this also to #199 which is closed.

Wanted to check on this again and to add some more information that I'm still observing.

Testing the Direct WFS for the Sea Regions theme, I'm seeing generated in the "before Period" test the following GET request.

https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns(xml,http://www.w3.org/XML/1998/namespace),xmlns(ns91,http://inspire.ec.europa.eu/schemas/sr/4.0),xmlns(wfs,http://www.opengis.net/wfs/2.0)&filter=<fes:Filter xmlns:ns91="http://inspire.ec.europa.eu/schemas/sr/4.0" xmlns:wfs="http://www.opengis.net/wfs/2.0" xmlns:fes="http://www.opengis.net/fes/2.0"><fes:Before><fes:ValueReference xmlns:tns="http://inspire.ec.europa.eu/schemas/sr/4.0">tns:beginLifespanVersion</fes:ValueReference><gml:TimePeriod xmlns:gml="http://www.opengis.net/gml/3.2" frame="http://www.iso.org/iso/iso8601" gml:id="TP01"><gml:beginPosition>**2017-07-12T17:40:00Z**</gml:beginPosition><gml:endPosition>**2017-07-12T18:20:00Z**</gml:endPosition></gml:TimePeriod></fes:Before></fes:Filter>

If I remove the "Z" zulu reference in the DateTime value, the request returns a record.

https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns(xml,http://www.w3.org/XML/1998/namespace),xmlns(ns91,http://inspire.ec.europa.eu/schemas/sr/4.0),xmlns(wfs,http://www.opengis.net/wfs/2.0)&filter=<fes:Filter xmlns:ns91="http://inspire.ec.europa.eu/schemas/sr/4.0" xmlns:wfs="http://www.opengis.net/wfs/2.0" xmlns:fes="http://www.opengis.net/fes/2.0"><fes:Before><fes:ValueReference xmlns:tns="http://inspire.ec.europa.eu/schemas/sr/4.0">tns:beginLifespanVersion</fes:ValueReference><gml:TimePeriod xmlns:gml="http://www.opengis.net/gml/3.2" frame="http://www.iso.org/iso/iso8601" gml:id="TP01"><gml:beginPosition>**2017-07-12T17:40:00**</gml:beginPosition><gml:endPosition>**2017-07-12T18:20:00**</gml:endPosition></gml:TimePeriod></fes:Before></fes:Filter>


So it seems like the URL request generation from the validator is expecting all DateTime values to be in a Zulu format, which in our case, we can't do because the data for our services are in the local time zone of the server the service is running. on. I've attached the test log file.

Please let me know if you have any questions. We'll try to keep this service running so you can test.

Thanks,
Robert

[DirectWFS_Sea_Regions_s0036.zip](https://github.com/opengeospatial/ets-wfs20/files/8774278/DirectWFS_Sea_Regions_s0036.zip)

**Comment by cportele** [comment]
@ghobona - Correct, if a time zone is specified in the predicate, the server has to take the timezone into account. Which is why the response of the server looks correct to me, although I have not analyzed this in detail. 

The feature properties are:

* `beginLifespanVersion`: `2017-07-12T17:20:00-07:00`
* `endLifespanVersion`: `2017-07-12T18:00:00-07:00`

Since the server does not return the timestamps with a timezone, i.e., the timestamps are in the local timezone of the server - in this case PDT. Now, if the client - in this case the CITE test - does not know the local timezone of the server, it cannot just assume that timestamps without a timezone are in UTC. The issue with the ETS is that it seems to make that assumption, which is invalid.

I think the only reasonable ETS approach would be to not use a timezone to filter temporal properties that are reported by a server without a time zone, because only then the test can assess, if the response is correct. If the server returns timestamps with a timezone, the ETS could also test proper timezone handling.

Here are some queries to demonstrate that the server seems to correctly handle timezone information:

* With implicit local timezone 
  * [beginLifespanVersion BEFORE 2017-07-12T17:30:00 => match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-12T17%3A30%3A00%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)
  * [beginLifespanVersion BEFORE 2017-07-12T17:15:00 => NO match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-12T17%3A15%3A00%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)
* With explicit local timezone
  * [beginLifespanVersion BEFORE 2017-07-12T17:30:00-07:00 => match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-12T17%3A30%3A00-07%3A00%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)
  * [beginLifespanVersion BEFORE 2017-07-12T17:15:00-07:00 => NO match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-12T17%3A15%3A00-07%3A00%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)
* With UTC and conversion to the local timezone
  * [beginLifespanVersion BEFORE 2017-07-13T00:30:00Z => match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-13T00%3A30%3A00Z%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)
  * [beginLifespanVersion BEFORE 2017-07-13T00:15:00Z => NO match](https://inspiredemo.esri.com/arcgis/rest/services/sr_test/MapServer/exts/InspireFeatureDownload/service?service=WFS&version=2.0.0&request=GetFeature&count=10&typenames=ns91:Sea&namespaces=xmlns%28xml,http://www.w3.org/XML/1998/namespace%29,xmlns%28ns91,http://inspire.ec.europa.eu/schemas/sr/4.0%29,xmlns%28wfs,http://www.opengis.net/wfs/2.0%29&filter=%3Cfes%3AFilter%20xmlns%3Ans91%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3ABefore%3E%3Cfes%3AValueReference%20xmlns%3Atns%3D%22http%3A%2F%2Finspire.ec.europa.eu%2Fschemas%2Fsr%2F4.0%22%3Etns%3AbeginLifespanVersion%3C%2Ffes%3AValueReference%3E%3Cgml%3ATimeInstant%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20frame%3D%22http%3A%2F%2Fwww.iso.org%2Fiso%2Fiso8601%22%20gml%3Aid%3D%22TP01%22%3E%3Cgml%3AtimePosition%3E2017-07-13T00%3A15%3A00Z%3C%2Fgml%3AtimePosition%3E%3C%2Fgml%3ATimeInstant%3E%3C%2Ffes%3ABefore%3E%3C%2Ffes%3AFilter%3E)

**Comment by ghobona** [comment]
Thanks @cportele . That is a fair point.

@pvretanos, let us know if you have a different view.

**Comment by dstenger** [comment]
@mlcharv We are currently working on the problem. Please check this issue for further updates.

**Comment by bpross-52n** [comment]
> @bpross-52n I have a similar issue with [this WFS](https://geodienste.komm.one/ows/services/org.506.78427a5b-2f82-4311-ba7d-74eb7b8c10f4_wfs/org.506.e21f771b-0d06-4dbc-b246-60aa8d12ed52?SERVICE=WFS&REQUEST=GetCapabilities&VERSION=2.0.0). ![grafik](https://user-images.githubusercontent.com/34163838/161543879-05d0ad5a-d7fd-4572-801b-1a05fb7487fe.png) There are two kinds of error messages:
> 
> 1. No temporal values found in results: property is "[http://inspire.ec.europa.eu/schemas/plu/4.0":validFrom](http://inspire.ec.europa.eu/schemas/plu/4.0%22:validFrom).  --> there is a validFrom value in the dataset in the SpatialPlan object (the OfficialDoument objects do not need one - see [UML](https://inspire.ec.europa.eu/data-model/approved/r4618-ir/html/EARoot/EA2/EA3/EA10/EA1/EA4/EA8445.htm)) and it is valid according to the INSPIRE Validator (dataset and report:
>    [validFrom.zip](https://github.com/opengeospatial/ets-wfs20/files/8409689/validFrom.zip) )
>    Could you please check why the error message is occurring?
> 2. java.lang.AssertionError: t1 (Period: 2012-04-04T00:00:00.000+0000/2012-04-04T23:59:59.999+0000) is not related to t2 (Period: 2012-04-04T16:19:59.999+0000/2012-04-05T00:59:59.998+0000): BEFORE. The actual temporal relation is OVERLAPS -> not sure what t1 and t2 are in this case and what would be invalid about them.
> 
> If you want me to open a separate issue on that, please let me know. But as one of the error messages is the same as in this issue, I thought it would make sense to add it here.
> 
> Thank you and all the best
> 
> Johanna

@JohannaOtt 
Regarding the first point: This is fixed on the beta instance of Teamengine.

Regarding the second point: This will be fixed by #241, see also here: [comment](https://github.com/opengeospatial/ets-wfs20/issues/226#issuecomment-1276195224)
